patch/0005-Support-Centec-TsingMa-SOC-MTD.patch

```diff
@@ -0,0 +1,406 @@
++	.quirks = SDHCI_QUIRK_CAP_CLOCK_BASE_BROKEN | SDHCI_QUIRK_BROKEN_ADMA,
++};
++
++static int sdhci_ctc5236_probe(struct platform_device *pdev)
++{
++	struct sdhci_host *host;
++	struct sdhci_pltfm_host *pltfm_host;
++	struct clk *clk;
++	int ret;
++
++	host = sdhci_pltfm_init(pdev, &sdhci_ctc5236_pdata, 0);
++	if (IS_ERR(host))
++		return PTR_ERR(host);
++
++	clk = devm_clk_get(&pdev->dev, "mmc_clk");
++	if (IS_ERR(clk)) {
++		dev_err(&pdev->dev, "Peripheral clk not found\n");
++		return PTR_ERR(clk);
++	}
++	pltfm_host = sdhci_priv(host);
++	pltfm_host->clk = clk;
++	clk_prepare_enable(clk);
++
++	regmap_base =
++	    syscon_regmap_lookup_by_phandle(pdev->dev.of_node, "ctc,sysctrl");
++	if (IS_ERR(regmap_base))
++		return PTR_ERR(regmap_base);
++
++	mmc_of_parse_voltage(pdev->dev.of_node, &host->ocr_mask);
++
++	ret = mmc_of_parse(host->mmc);
++	if (ret)
++		goto err_sdhci_add;
++
++	ret = sdhci_add_host(host);
++	if (ret)
++		goto err_sdhci_add;
++	return 0;
++
++err_sdhci_add:
++	sdhci_pltfm_free(pdev);
++	return ret;
++}
++
++static const struct of_device_id sdhci_ctc5236_of_match[] = {
++	{.compatible = "centec,ctc5236-sdhci"},
++	{},
++};
++
++MODULE_DEVICE_TABLE(of, sdhci_ctc5236_of_match);
++
++static struct platform_driver sdhci_ctc5236_driver = {
++	.driver = {
++		   .name = "sdhci-ctc5236",
++		   .pm = &sdhci_pltfm_pmops,
++		   .of_match_table = of_match_ptr(sdhci_ctc5236_of_match),
++		   },
++	.probe = sdhci_ctc5236_probe,
++	.remove = sdhci_pltfm_unregister,
++};
++
++module_platform_driver(sdhci_ctc5236_driver);
++
++MODULE_DESCRIPTION("SDHCI driver for Centec TsingMa SoCs");
++MODULE_AUTHOR("Jay Cao <caoj@centecnetworks.com>");
++MODULE_LICENSE("GPL v2");
+diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
+index 270d3c9..76b3aa9 100644
+--- a/drivers/mtd/devices/m25p80.c
++++ b/drivers/mtd/devices/m25p80.c
+@@ -299,7 +299,7 @@ static const struct spi_device_id m25p_ids[] = {
+ 	{"m25p40"},	{"m25p80"},	{"m25p16"},	{"m25p32"},
+ 	{"m25p64"},	{"m25p128"},
+ 	{"w25x80"},	{"w25x32"},	{"w25q32"},	{"w25q32dw"},
+-	{"w25q80bl"},	{"w25q128"},	{"w25q256"},
++	{"w25q80bl"},	{"w25q128"},	{"w25q256"},	{"mx25u3235f"},
+ 
+ 	/* Flashes that can't be detected using JEDEC */
+ 	{"m25p05-nonjedec"},	{"m25p10-nonjedec"},	{"m25p20-nonjedec"},
+diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
+index f028277..c18b897 100644
+--- a/drivers/mtd/spi-nor/spi-nor.c
++++ b/drivers/mtd/spi-nor/spi-nor.c
+@@ -1095,6 +1095,7 @@ static const struct flash_info spi_nor_ids[] = {
+ 	{ "mx66u51235f", INFO(0xc2253a, 0, 64 * 1024, 1024, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
+ 	{ "mx66l1g45g",  INFO(0xc2201b, 0, 64 * 1024, 2048, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+ 	{ "mx66l1g55g",  INFO(0xc2261b, 0, 64 * 1024, 2048, SPI_NOR_QUAD_READ) },
++	{ "mx25u3235f",  INFO(0xc22539, 0, 64 * 1024, 512, SECT_4K) },
```

Comment:
can you use this patch instead?

https://github.com/torvalds/linux/commit/748df6d831871ca2338644e6c61a84a02a8e2786